graph/db/sql_migration_test.go

```diff
@@ -38,6 +67,44 @@ func TestMigrateGraphToSQL(t *testing.T) {
 		{
 			name: "empty",
 		},
+		{
+			name:  "nodes",
+			write: writeUpdate,
+			//nolint:ll
+			objects: []any{
+				// Normal node with all fields.
+				makeTestNode(t),
+				// A node with no node announcement.
+				makeTestShellNode(t),
+				// A node with an announcement but no addresses.
+				makeTestNode(t, func(n *models.LightningNode) {
+					n.Addresses = nil
+				}),
+				// A node with all types of addresses.
+				makeTestNode(t, func(n *models.LightningNode) {
+					n.Addresses = []net.Addr{
+						testAddr,
+						testIPV4Addr,
+						testIPV6Addr,
+						anotherAddr,
+						testOnionV2Addr,
+						testOnionV3Addr,
+						testOpaqueAddr,
+					}
+				}),
+				// No extra opaque data.
+				makeTestNode(t, func(n *models.LightningNode) {
+					n.ExtraOpaqueData = nil
```

Comment:
nit: iiuc this won't test the case where `marshalExtraOpaqueData` fails. I'd suggest we add one too so we can test that it's indeed skipping nodes as expected.

Reply:
this is covered in `TestSQLMigrationEdgeCases` below

Reply:
Ah yes, sorry I totally overlooked that!